runtime/lua/vim/lsp.lua

```diff
@@ -1255,27 +1255,30 @@ function lsp._vim_exit_handler()
       send_kill = true
       timeouts[client_id] = timeout
       max_timeout = math.max(timeout, max_timeout)
-    else
-      active_clients[client_id] = nil
     end
   end
 
   local poll_time = 50
 
   local function check_clients_closed()
```

Comment:
This won't have the desired effect, because the function no longer considers whether the client is actually active, so a client that shuts down early but has a long timeout will still block.

Reply:
> You would still want to filter on active clients.

That's what I'm doing, isn't it?

This function only returns false if it finds an active client, whose timeout has not yet expired.

Probably I've misunderstood; but you're going to have to spell out for me what I've missed, please!

Reply:
Sorry, I'm confused a bit, where are you nil-ing timeouts[client_id]? Shouldn't this never be nil due to the if_nil check I added?

Reply:
It's `nil` if the `exit_timeout` for a client is false, because `timeouts[client_id]` is only set if it's not false.

(I think that's right, lua noob here though so willing to be corrected)

Reply:
nil != false in lua

Reply:
No, but false = false.

So if I explicitly set `exit_timeout = false`, the timeout for a client will be left `nil`, right?

Reply:
Yes, I think it's fine. I think it's a bit of a confusing holdover that I introduced but I see what you've done now.